core/src/main/java/com/google/googlejavaformat/java/Formatter.java

```diff
@@ -203,6 +204,25 @@ public String formatSource(String input) throws FormatterException {
   }
 
   /**
+   * Format an input string (a Java compilation unit) and optimize imports into an output string.
+   *
+   * @param input the input string
+   * @param joi if not {@code null}, the imports are optimized accordingly
+   * @return the output string
+   * @throws FormatterException if the input string cannot be parsed
+   */
+  public String formatSource(String input, JavadocOnlyImports joi) throws FormatterException {
+    String output = formatSource(input, Collections.singleton(Range.closedOpen(0, input.length())));
```

Comment:
```
org.junit.ComparisonFailure: bad output for B21647014  <Click to see difference>
org.junit.ComparisonFailure: bad output for B21647014  <Click to see difference>
org.junit.ComparisonFailure: bad output for B21647014  <Click to see difference>
org.junit.ComparisonFailure: bad output for B21647014  <Click to see difference>

org.junit.ComparisonFailure: bad output for M  <Click to see difference>
org.junit.ComparisonFailure: bad output for M  <Click to see difference>
org.junit.ComparisonFailure: bad output for M  <Click to see difference>
org.junit.ComparisonFailure: bad output for M  <Click to see difference>

org.junit.ComparisonFailure: bad output for Unformatted  <Click to see difference>
org.junit.ComparisonFailure: bad output for Unformatted  <Click to see difference>
org.junit.ComparisonFailure: bad output for Unformatted  <Click to see difference>
org.junit.ComparisonFailure: bad output for Unformatted  <Click to see difference>

org.junit.ComparisonFailure:  <Click to see difference>
	at com.google.common.truth.Platform.comparisonFailure(Platform.java:133)
	at com.google.common.truth.Truth$1.failComparing(Truth.java:77)
	at com.google.common.truth.StringSubject.isEqualTo(StringSubject.java:69)
	at com.google.googlejavaformat.java.FormatterTest.importsNotReorderedByDefault(FormatterTest.java:249)


org.junit.ComparisonFailure:  <Click to see difference>
	at com.google.googlejavaformat.java.JavadocFormattingTest.doFormatTest(JavadocFormattingTest.java:1269)
	at com.google.googlejavaformat.java.JavadocFormattingTest.brAtSignBug(JavadocFormattingTest.java:823)
```
The initial blocks of four failures are from `FormatterIntegrationTest.idempotent...` parameterized test.

I vote for keeping the new method name for the new behaviour.